src/synapse/api.py

```diff
@@ -182,15 +192,26 @@ def get_version() -> str:
         session.mount("http://", HTTPAdapter(max_retries=retries))
         res = session.get(VERSION_URL, timeout=10)
         res.raise_for_status()
-        server_version = res.json()["server_version"]
-        version_match = re.search(r"^([^\s(]+)", server_version)
-        if version_match:
-            return version_match.group(1)
-        return server_version
-    except (
-        requests.exceptions.ConnectionError,
-        requests.exceptions.Timeout,
-        requests.exceptions.HTTPError,
-    ) as exc:
-        logger.exception("Failed to request %s : %r", VERSION_URL, exc)
-        raise NetworkError(f"Failed to request {VERSION_URL}.") from exc
+        res_json = res.json()
+        server_version = res_json.get("server_version", None)
```

Comment:
What about the case where `res.json()` is not a dict?

Reply:
Changed to check in this call and also the others.